client/templates/courses/tagged-courses.js

```diff
@@ -6,3 +6,11 @@ Template.taggedCourses.helpers({
         return Session.get('tag');
     }
 });
+
+Template.taggedCourses.onCreated(function(){
+  var controller = Iron.controller();
+  var params = controller.getParams();
+  this.autorun(function(){
```

Comment:
Remove `autorun()`. 

The `autorun()` is used for code that should re-run when reactive data sources, such as database cursors (`Collection.find()`) change. Since there is no reactive data source within your `autorun()` function, there is no need for `autorun()` at all.